test/integration/targets/uri/tasks/proxy.yml

```diff
@@ -0,0 +1,96 @@
+- name: Get groups
+  ansible.builtin.getent:
+    database: group
+
+- name: Set var for nobody/nogroup group
+  set_fact:
+    nobody_group: '{{ ansible_facts.getent_group|list|select("match", "no(group|body)")|first }}'
+
+- name: Install tinyproxy config
+  copy:
+    dest: '{{ remote_tmp_dir }}/tinyproxy.conf'
+    content: |
+      User nobody
+      Group {{ nobody_group }}
+      Port 8080
+      Listen 127.0.0.1
+      Timeout 10
+      LogLevel Info
+      MaxClients 1
+      StartServers 1
+      Allow 127.0.0.1
+      Allow ::1
+      Allow ::
+      ViaProxyName "tinyproxy"
+      # LogFile "/tmp/tinyproxy.log"
+
+- name: Generate cert for stunnel
+  command: >-
+    openssl req -new -x509 -days 365 -nodes -out stunnel.pem -keyout stunnel.pem \
```

Comment:
Based on the low-level TLS management I see here I believe that my suggestion to use `trustme` might simplify the configuration and positively affect the maintainability: https://github.com/ansible/ansible/pull/83089/files#r1571440157.